lib/game.rb

```diff
@@ -0,0 +1,102 @@
+class Game < Board
+  attr_accessor :board, :player_1, :player_2
+WIN_COMBINATIONS = [
+[0,1,2],
+[3,4,5],
+[6,7,8],
+[0,4,8],
+[2,4,6],
+[0,3,6],
+[1,4,7],
+[2,5,8]
+]
+
+  def initialize (player_1 = nil, player_2 = nil, board = nil)
+    @player_1 = player_1 || @player_1 = Players::Human.new("X")
+    @player_2 = player_2 || @player_2 = Players::Human.new("O")
+    @board = board || @board = Board.new
+  end
+  
+  def current_player
+    if @board.turn_count.even?
+      @player_1
+    else 
+      @player_2
+   end 
+  end 
+  
+  def won?
+    WIN_COMBINATIONS.detect do |win| 
+      @board.cells[win[0]] == @board.cells[win[1]] && @board.cells[win[0]] == @board.cells[win[2]] && (@board.cells[win[0]] == "X" || @board.cells[win[0]] == "O")
+    end
+  end
+  
+  def draw?
+    !won? && @board.full?
+  end
+  
+  def over?
+     won? || draw?
+  end
+  
+  def winner 
+    if won? == nil
+      nil
+    else
+      @board.cells[won?[0]]
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.